httpcore/src/main/java/org/apache/http/Consts.java

```diff
@@ -42,6 +42,8 @@
     public static final int HT = 9;  // <US-ASCII HT, horizontal-tab (9)>
 
     public static final Charset UTF_8 = Charset.forName("UTF-8");
+    public static final Charset UTF_16 = Charset.forName("UTF-16");
+    public static final Charset UTF_32 = Charset.forName("UTF-32");
```

Comment:
FYI, UTF-32 has not been included in Java 7's `StandardCharsets`.

AFAIK all Java installations support it, but I could better handle that error if missing UTF-32 support is a concern.